Basic manipulations of phyloseq objects. Rarefying sample species tables (or sample approximate sequence variant/operational taxonomic unit tables),
subsetting sample data (by time or data type), calculating distance metrics, phylogenetic analyses, ordinations. Does not include the creation of phyloseq
object or incorporation of meta-data (see https://github.com/TonyMane/caddis/blob/main/tutorial_dada2_to_phyloseq.md). For background on phyloseq  see
https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0061217 or https://joey711.github.io/phyloseq/). 

```
library(phyloseq)
library(ape)
load("caddis03282022.rds")
```
The rds file should be very simple, containing a sinlge object. 
Can just type 'objects()' do see see whats been loaded.

```
objects()
[1] "ps"
```
Some basic data about the object can be printed by typing its name.

```
ps
```
And below is what should be printed to the screen.
```
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 69382 taxa and 219 samples ]
sample_data() Sample Data:       [ 219 samples by 4 sample variables ]
tax_table()   Taxonomy Table:    [ 69382 taxa by 6 taxonomic ranks ]
refseq()      DNAStringSet:      [ 69382 reference sequences ]
```
The'otu_table()' is actually an approximate sequence variant table. For background about how these were generated see: 
https://www.nature.com/articles/nmeth.3869.
For some info on why aproximate sequence variants (ASVs) are used exclusively for 16S rRNA gene analyses, see:
https://www.nature.com/articles/ismej2017119. 

The 'otu_table()' is orientated with ASVs in columns and samples in rows. 
We can see the sequencing depth per sample by using 'colSums' on the transposed 'otu_table'
```
colSums(t(otu_table(ps))) #the whole output will print to screen.
```
You can see there is quite a bit of variability in sequencing depth, but thousands of reads per sample is not uncommon. 
More on this later

Each 'taxa' is actually an ASV. So there are 69382 ASVs distributed throughout 219 samples. The 'tax_table()' has information
on each ASV and its evotionary origins, or taxonomic identity. This was generated by comparing each ASV to a database of 16S rRNA genes whose 
taxonomy has been evaluated phylogenically (the SILVA rRNA gene database). 

We can look at this data a variety of ways. First, manually:
```
Taxonomy Table:     [6 taxa by 6 taxonomic ranks]:
     Kingdom    Phylum          Class              Order         Family Genus
ASV1 "Bacteria" "Cyanobacteria" "Oxyphotobacteria" "Chloroplast" NA     NA   
ASV2 "Bacteria" "Cyanobacteria" "Oxyphotobacteria" "Chloroplast" NA     NA   
ASV3 "Bacteria" "Cyanobacteria" "Oxyphotobacteria" "Chloroplast" NA     NA   
ASV4 "Bacteria" "Cyanobacteria" "Oxyphotobacteria" "Chloroplast" NA     NA   
ASV5 "Bacteria" "Cyanobacteria" "Oxyphotobacteria" "Chloroplast" NA     NA   
ASV6 "Bacteria" "Cyanobacteria" "Oxyphotobacteria" "Chloroplast" NA     NA
```
As you can see, the top 6 ASVs are all chloroplast sequences. These sequences are not associated with single celled bacteria and archaea (i.e. 'microbiome'), but multicellular aquatic chloroplasts. Its considered good practice to remove these and sequences associated with mitochondria. 
Phyloseq has some nice utilities for easy access to sequences, their IDs, and subseqent removal. We can remove all "Chloroplasts" and "Mitochondria" using the below commands:
```
ps_clean = subset_taxa(ps, Order != "Chloroplast")
```
And then have a look at the top of the 'cleaned' data.

```
head(tax_table(ps_clean))
Taxonomy Table:     [6 taxa by 6 taxonomic ranks]:
      Kingdom    Phylum           Class                 Order                   Family             Genus                    
ASV10 "Bacteria" "Cyanobacteria"  "Oxyphotobacteria"    "Nostocales"            "Phormidiaceae"    "Tychonema_CCAP_1459-11B"
ASV11 "Bacteria" "Bacteroidetes"  "Bacteroidia"         "Chitinophagales"       "Saprospiraceae"   NA                       
ASV17 "Bacteria" "Cyanobacteria"  "Oxyphotobacteria"    "Nostocales"            "Phormidiaceae"    "Tychonema_CCAP_1459-11B"
ASV23 "Bacteria" "Bacteroidetes"  "Bacteroidia"         "Chitinophagales"       "Saprospiraceae"   NA                       
ASV26 "Bacteria" "Proteobacteria" "Gammaproteobacteria" "Betaproteobacteriales" "Burkholderiaceae" "Rhodoferax"             
ASV27 "Bacteria" "Bacteroidetes"  "Bacteroidia"         "Chitinophagales"       "Saprospiraceae"   NA
```
Then perform another removal of mitochondria:

```
ps_clean = subset_taxa(ps_clean, Class != "Mitochondria")
```
The chloroplast/mitochondria free phyloseq object should read as such:

```
ps_clean
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 53124 taxa and 219 samples ]
sample_data() Sample Data:       [ 219 samples by 4 sample variables ]
tax_table()   Taxonomy Table:    [ 53124 taxa by 6 taxonomic ranks ]W
refseq()      DNAStringSet:      [ 53124 reference sequences ]
```
We now want to adjust each of the sample sizes such they are equal, a process often referred to as rarefaction. 
Note, whether to rarefy or not rarefy is the subject of much debate in microbiome research. See:
https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1003531 
And https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btac127/6536959?login=true.

We'll do it here, mainly because we aren't as concerned with 'rare' taxa, but we'll also come back to this later. 
To rarefy, phyloseq has an easy command (rarefy_even_depth):

```
ps_rare = ps_rare = rarefy_even_depth(ps_clean, 1000)
```
We would like to do some comparisons between samples, looking at how similar (or distant) each of the samples is to one another.
There a LOT of ways to do this, commonly termed Beta diversity. 
A very popular method is Unifrac. See :
https://www.nature.com/articles/ismej2010133

To perform unifrac on our matrix we need to first generate a phylogenetic tree and incorporate it into the phyloseq object.

```
tree = rtree(ntaxa(ps.rare), rooted=TRUE, tip.label=taxa_names(ps.rare))
ps_rare = merge_phyloseq(ps_rare, tree)
```
Now we can run unifrac.

```
ps_rare_unifrac = phyloseq :: distance(ps_rare, "unifrac")
```
This produces a large rectangular distance matrix. We do a lot with this, but a common method of analysis is ordination, a quick way to visually
explore the data. I use the vegan package for this, there might be ones available directly in phyloseq though. For three dimensional ordination use k=3.

```
library(vegan)
ps_rare_unifrac_nmds = monoMDS(ps_rare_unifrac, k=3)
```

We can then plot the result with 'scatterplot3d'.
```
library(scatterplot3d)
scatterplot3d(ps_rare_unifrac.nmds$points[,1:3], color=sample_data(ps_rare)$Color, pch=16)
```
