Basic manipulations of phyloseq objects. Rarefying sample species tables (or sample approximate sequence variant/operational taxonomic unit tables),
subsetting sample data (by time or data type), calculating distance metrics, phylogenetic analyses, ordinations. Does not include the creation of phyloseq
object or incorporation of meta-data (see https://github.com/TonyMane/caddis/blob/main/tutorial_dada2_to_phyloseq.md). For background on phyloseq  see
https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0061217 or https://joey711.github.io/phyloseq/). 

```
library(phyloseq)
library(ape)
load("caddis03282022.rds")
```
The rds file should be very simple, containing a sinlge object. 
Can just type 'objects()' do see see whats been loaded.

```
objects()
[1] "ps"
```
Some basic data about the object can be printed by typing its name.

```
ps
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 69382 taxa and 219 samples ]
sample_data() Sample Data:       [ 219 samples by 4 sample variables ]
tax_table()   Taxonomy Table:    [ 69382 taxa by 6 taxonomic ranks ]
refseq()      DNAStringSet:      [ 69382 reference sequences ]
```
The'otu_table()' is actually an approximate sequence variant table. For background about how these were generated see: 
https://www.nature.com/articles/nmeth.3869.
For some info on why aproximate sequence variants (ASVs) are used exclusively for 16S rRNA gene analyses, see:
https://www.nature.com/articles/ismej2017119. 

The 'otu_table()' is orientated with ASVs in columns and samples in rows. 
We can see the sequencing depth per sample by using 'colSums' on the transposed 'otu_table'
```
colSums(t(otu_table(ps))) #the whole output will print to screen.
```
You can see there is quite a bit of variability in sequencing depth, but thousands of reads per sample is not uncommon. 
More on this later

Each 'taxa' is actually an ASV. So there are 69382 ASVs distributed throughout 219 samples. The 'tax_table()' has information
on each ASV and its evotionary origins, or taxonomic identity. This was generated by comparing each ASV to a database of 16S rRNA genes whose 
taxonomy has been evaluated phylogenically (the SILVA rRNA gene database). 

We can look at this data a variety of ways. First, manually:
```
Taxonomy Table:     [6 taxa by 6 taxonomic ranks]:
     Kingdom    Phylum          Class              Order         Family Genus
ASV1 "Bacteria" "Cyanobacteria" "Oxyphotobacteria" "Chloroplast" NA     NA   
ASV2 "Bacteria" "Cyanobacteria" "Oxyphotobacteria" "Chloroplast" NA     NA   
ASV3 "Bacteria" "Cyanobacteria" "Oxyphotobacteria" "Chloroplast" NA     NA   
ASV4 "Bacteria" "Cyanobacteria" "Oxyphotobacteria" "Chloroplast" NA     NA   
ASV5 "Bacteria" "Cyanobacteria" "Oxyphotobacteria" "Chloroplast" NA     NA   
ASV6 "Bacteria" "Cyanobacteria" "Oxyphotobacteria" "Chloroplast" NA     NA
```
